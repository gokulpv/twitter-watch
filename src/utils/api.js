const BACKEND_BASE_URL = "http://localhost:3000";

export const get = async (pathname) => {
  const response = await fetch(`${BACKEND_BASE_URL}/${pathname}`);
  const data = await response.json();
  if (!response.ok) {
    // Can be made to throw specific error messages
    throw new Error("Some error occured.Please check the url");
  }
  return data;
};
