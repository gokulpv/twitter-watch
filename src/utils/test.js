export const findByTestId = (component, id) => {
  const wrapper = component.find(`[data-testid='${id}']`);
  return wrapper;
};
