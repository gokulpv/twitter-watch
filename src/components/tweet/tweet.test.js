import React from "react";
import Tweet from "./Tweet";
import { shallow } from "enzyme";
import { findByTestId } from "../../utils/test";
import Date from "./date/Date";
import Media from "./media/media";

const mockProps = {
  created_at: "Tue Sep 28 20:29:31 +0000 2021",
  id_str: "1442949613971935233",
  full_text: "This is a test",

  entities: {
    hashtags: [],
    symbols: [],
    user_mentions: [],
    urls: [],
    media: [
      {
        media_url: "4487257089/img/csZRMkgU7c6nRv33.jpg",
      },
    ],
  },
};

describe("Tweet component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Tweet tweet={mockProps} />);
  });

  // Test for render without any errors
  it("should render Tweet component", () => {
    expect(component).toMatchSnapshot();
  });

  it("find the text content to be the same as mocked", () => {
    expect(findByTestId(component, "tweet-text").text()).toMatch(
      /This is a test/
    );
  });
  it("renders a <Date /> components", () => {
    expect(component.find(Date).length).toBe(1);
  });
  it("renders a <Media /> component as per the given prop", () => {
    expect(component.find(Media).length).toBe(1);
  });
});
