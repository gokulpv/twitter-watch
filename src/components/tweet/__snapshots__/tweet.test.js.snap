// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet component should render Tweet component 1`] = `ShallowWrapper {}`;
